### ISSUE TITLE
output dimension mistmatch when using sparse_categorical_crossentropy

### ISSUE THREAD

**Issue by TQuy** [keras-team/keras#13292]
I am trying to implemenent word-level seq2seq using the keras example example from keras `https://keras.io/examples/lstm_seq2seq/`
```
### ENCODER
lstm_units = 128
encoder_input = Input(shape = (None,), name = 'Encoder_input')
embedding = Embedding(input_dim = len(compact_vocab), output_dim = 30, name = 'Embedding')
encoder_embedding = embedding(encoder_input)
encoder = Bidirectional(LSTM(lstm_units, return_state = True, go_backwards = True, name = 'Encoder'))
_, forward_h, forward_c, backward_h, backward_c = encoder(encoder_embedding)
state_h = Concatenate()([forward_h, backward_h])
state_c = Concatenate()([forward_c, backward_c])
encoder_state = [state_h, state_c]
### DECODER
decoder_input = Input(shape = (None,), name = 'Decoder_input')
# decoder_embedding = Embedding(input_dim = len(compact_vocab), output_dim = 30, name = 'decoder_embedding')(decoder_input)
decoder_embedding = embedding(decoder_input)
decoder = LSTM(lstm_units*2, return_state = True, return_sequences = True, name = 'Decoder')
decoder_output, _, _ = decoder(decoder_embedding, initial_state = encoder_state)
decoder_dense = TimeDistributed(Dense(len(compact_vocab), activation = 'softmax'), name = 'Decoder_output_layer')
decoder_output_vector = decoder_dense(decoder_output)
model = Model([encoder_input, decoder_input], decoder_output_vector)
model.summary()
```
![image](https://user-images.githubusercontent.com/45731304/64307636-eb4eef00-cfc0-11e9-9df8-cde741183150.png)
When I am trying to compile the model
`model.compile(optimizer='Adadelta',
            loss='sparse_categorical_crossentropy',
            metrics=['sparse_categorical_accuracy'])
checkpointer = ModelCheckpoint("model/s2s_chatbot.h5", monitor = 'loss', verbose= 1, save_weights_only=False)
model.fit_generator(train_generator, 
                    steps_per_epoch = steps_per_epoch_train,
                    epochs = 1,
                    verbose = 1,
                   callbacks=[checkpointer])`
I face this error
![image](https://user-images.githubusercontent.com/45731304/64307902-b3947700-cfc1-11e9-83a0-918f6563bd77.png)
When I try to compile the character-level seq2seq, I also face this issue, and I have to convert the expected output to one-hot vector

**Comment by deepakmeena635** [comment]
same issue, is a bug most probably
